### PR TITLE
Refactor time text positioning

### PR DESCRIPTION
**Pull request by yschimke**
#### WHAT

Refactor time text positioning.

#### WHY

To allow for animating between states.  No change here.

#### Checklist :clipboard:
- [ ] Add explicit visibility modifier and explicit return types for public declarations
- [ ] Run spotless check
- [ ] Run tests
- [ ] Update metalava's signature text files

### CODE REVIEW

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt [hunk 1]
```diff
@@ -114,3 +119,23 @@ public fun Modifier.scrollAway(
...
+        is ScalingLazyColumnScrollableState -> {
+            val offsetDp = with(LocalDensity.current) {
+                state.initialOffsetPx.toDp()
+            }
+            this.scrollAway(state.scalingLazyListState, state.initialIndex, offsetDp)
+        }
+        is ScalingLazyListState -> this.scrollAway(state)
+        is LazyListState -> this.scrollAway(state)
+        is ScrollState -> this.scrollAway(state)
+        null -> this.hidden()
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:136]: nit: redundant given the else statement below, unless you'd like it to be explicit, then maybe add a comment explaining why?

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt [hunk 2]
```diff
@@ -114,3 +119,23 @@ public fun Modifier.scrollAway(
...
+    when (val state = scrollState.value) {
+        is ScalingLazyColumnScrollableState -> {
+            val offsetDp = with(LocalDensity.current) {
+                state.initialOffsetPx.toDp()
+            }
+            this.scrollAway(state.scalingLazyListState, state.initialIndex, offsetDp)
+        }
+        is ScalingLazyListState -> this.scrollAway(state)
+        is LazyListState -> this.scrollAway(state)
+        is ScrollState -> this.scrollAway(state)
```
Comment (luizgrp) [inline compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:135]: nit: separate common behaviour with commas 
```
is ScalingLazyListState, 
is LazyListState, 
is ScrollState -> this.scrollAway(state)
```

**Review by luizgrp — APPROVED** [review]

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt (hunk 1 above)
Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:136]: Will do.

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt (hunk 2 above)
Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:135]: It's not common behaviour, it's different overloads of the same method name.

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt (hunk 1 above)
Comment (yschimke) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:137]: Actually found a bug.

File: compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt (hunk 2 above)
Comment (luizgrp) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:135]: yeah my wording doesn't sound right :D

Comment (luizgrp) [reply compose-layout/src/main/java/com/google/android/horologist/compose/layout/ScrollAway.kt:135]: oh I see what you mean, hard to spot in code review! 👍

**Pull request merged**